jni/src/Foreign/JNI/Types.hs

```diff
@@ -134,21 +134,21 @@ instance IsReferenceType ('Array ty)
 instance IsReferenceType ty => IsReferenceType ('Generic ty tys)
 
 data instance Sing (a :: JType) where
-  SClass :: String -> Sing ('Class sym)
-  SIface :: String -> Sing ('Iface sym)
-  SPrim :: String -> Sing ('Prim sym)
+  SClass :: JNI.String -> Sing ('Class sym)
+  SIface :: JNI.String -> Sing ('Iface sym)
+  SPrim :: JNI.String -> Sing ('Prim sym)
```

Comment:
This is going to reintroduce performance problems. Can't it be left as is? https://github.com/tweag/inline-java/issues/11